### PR TITLE
Current sense amplifier driver

### PR DESCRIPTION
**Pull request by nixward**
Add a current sense amplifier driver.

Implements: https://github.com/zephyrproject-rtos/zephyr/issues/60415

### CODE REVIEW

File: dts/bindings/sensor/zephyr,current.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+properties:
+  io-channels:
+    required: true
+    description: |
+      Channels available with this divider configuration
+
+  shunt-r:
+    type: int
+    required: true
+    description: Shunt resistance in milli Ohms
```
Comment (jeppenodgaard) [inline dts/bindings/sensor/zephyr,current.yaml:20]:
```suggestion
    description: Shunt resistance in milliohms
```

**Review by jeppenodgaard — APPROVED** [review]
Minor comments otherwise LGTM!

File: drivers/sensor/current/current.c [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+	raw_val *= ACCUR_FACTOR;
+
+	ret = adc_raw_to_millivolts_dt(&config->adc, &raw_val);
+	if (ret != 0) {
+		LOG_ERR("raw_to_mv: %d", ret);
+		return ret;
+	}
+
+	/* micro Amp */
```
Comment (jeppenodgaard) [inline drivers/sensor/current/current.c:69]:
```suggestion
	/* microamp */
```

**Review by marco85mars20 — APPROVED** [review]

**Comment by nixward** [comment]
I've added the DNM label so I can update the driver to use the new current sensor binding when I get some spare time.

File: drivers/sensor/CMakeLists.txt [hunk 3]
```diff
@@ -25,6 +25,7 @@ add_subdirectory_ifdef(CONFIG_BMI323 bmi323)
 add_subdirectory_ifdef(CONFIG_BMM150 bmm150)
 add_subdirectory_ifdef(CONFIG_BMP388 bmp388)
 add_subdirectory_ifdef(CONFIG_BQ274XX bq274xx)
+add_subdirectory_ifdef(CONFIG_CURRENT_AMP current_amp)
 add_subdirectory_ifdef(CONFIG_CCS811 ccs811)
```
Comment (MaureenHelm) [inline drivers/sensor/CMakeLists.txt:29]: Not quite alpha order, please move down a line

File: drivers/sensor/current_amp/current_amp.c [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+	return 0;
+}
+
+#define CURRENT_SENSE_AMPLIFIER_INIT(inst)                                                         \
+	static struct current_sense_amplifier_data current_amp_##inst##_data;                      \
+                                                                                                   \
+	static const struct current_sense_amplifier_dt_spec current_amp_##inst##_config =          \
+		CURRENT_SENSE_AMPLIFIER_DT_SPEC_GET(DT_DRV_INST(inst));                            \
+                                                                                                   \
+	DEVICE_DT_INST_DEFINE(inst, &current_init, NULL, &current_amp_##inst##_data,               \
```
Comment (MaureenHelm) [inline drivers/sensor/current_amp/current_amp.c:111]: Use `SENSOR_DEVICE_DT_INST_DEFINE`

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
Please add this sensor to `tests/drivers/build_all/sensor/adc.dtsi`

**Comment by MaureenHelm** [comment]
> Please add this sensor to `tests/drivers/build_all/sensor/adc.dtsi`

Oops, I see now that you already did this in the second commit

File: drivers/sensor/current_amp/current_amp.c (hunk 4 above)
Comment (nixward) [reply drivers/sensor/current_amp/current_amp.c]: Done

**Review by MaureenHelm — DISMISSED** [review]

**Review by marco85mars20 — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@teburd @yperess @avisconti please take a look

File: drivers/sensor/current_amp/Kconfig [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+#
+# Copyright (c) 2023 FTP Technologies
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config CURRENT_AMP
+	bool "Current sense amplifier driver"
+	default y
+	depends on DT_HAS_CURRENT_SENSE_AMPLIFIER_ENABLED
+	select ADC
```
Comment (nandojve) [inline drivers/sensor/current_amp/Kconfig:11]:
```diff
-	select ADC
+	depends on ADC
```

**Review by nandojve — CHANGES_REQUESTED** [review]

File: drivers/sensor/current_amp/Kconfig (hunk 5 above)
Comment (nixward) [reply drivers/sensor/current_amp/Kconfig]: Thanks, fixed

**Review by nandojve — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**